WorkloadManagementSystem/Utilities/test/Test_PilotCStoJSONSynchronizer.py

```diff
@@ -173,9 +174,19 @@ def test_syncchecksum(self):
     synchroniser._syncChecksum()
     assert self.testCfgFileName in synchroniser._checksumDict
     assert synchroniser._checksumDict[self.testCfgFileName] == expectedHash
-    assert open('checksums.sha512', 'rb').read().split('\n')[0] == '%s  %s' % (expectedPJHash, 'pilot.json'), \
-        'pilot.json content: ' + repr(open('pilot.json', 'rb').read())
-    assert open('checksums.sha512', 'rb').read().split('\n')[1] == '%s  %s' % (expectedHash, self.testCfgFileName)
+    with open('checksums.sha512', 'rt') as fp:
+      checksums = fp.read()
+    with open('pilot.json', 'rt') as fp:
+      pilot_json_content = fp.read()
+
+    if not six.PY2:
```

Comment:
Why this just for python3? Is json in python2 deterministic?
Did this fail without re-calculation?

Reply:
Neither is deterministic and it's an inherent "feature" of JSON specification. It's probably stable for a given Python version but it shouldn't be relied upon given JSON explicitly allows reading and writing to shuffle the order of maps. 

The reason I only applied this for Python 3 is that the test already works for Python 2 so I don't want to break anything that is already working. Next time the hashes change we should probably drop the hard coded values.

Reply:
But then it could be someone breaks the pilot.json writing and we just happily calculate the checksum for the broken file.

Reply:
That's why I left it as-is for Python 2. If the name is to be believed (`test_syncchecksum`) this test is only validating the checksumming but 🤷.

Reply:
two tests for the price of one!